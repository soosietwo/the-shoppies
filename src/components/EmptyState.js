import React from "react";
import { EmptyState } from "@shopify/polaris";

const EmptyStateMessage = (movies) => {
  return !movies.length ? (
    <EmptyState
      heading="No movies found"
      action=""
      image="https://cdn.shopify.com/s/files/1/2376/3301/products/emptystate-files.png"
    >
      <p>Search to get started!</p>
    </EmptyState>
  ) : undefined;
};

export default EmptyStateMessage;
