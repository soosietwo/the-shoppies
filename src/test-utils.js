// test-utils.js
import React from "react";
import { render } from "@testing-library/react";
import { AppProvider, Frame } from "@shopify/polaris";
import en from "@shopify/polaris/locales/en.json";
import Store from "./store";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const AllTheProviders = ({ children }) => {
  return (
    <AppProvider i18n={en}>
      <Store>
        <Frame>{children}</Frame>
      </Store>
    </AppProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
