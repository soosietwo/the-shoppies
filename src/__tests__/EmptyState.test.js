import React from "react";
import { render } from "../test-utils";
import EmptyState from "../components/EmptyState";

describe("EmptyState", () => {
  it("renders and matches snapshot", () => {
    const { container } = render(<EmptyState movies={[]} />);

    expect(container.querySelector(".Polaris-EmptyState")).toMatchSnapshot();
  });
});
