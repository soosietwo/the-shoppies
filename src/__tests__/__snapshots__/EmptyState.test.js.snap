// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState renders and matches snapshot 1`] = `
<div
  class="Polaris-EmptyState Polaris-EmptyState--withinContentContainer"
>
  <div
    class="Polaris-EmptyState__Section"
  >
    <div
      class="Polaris-EmptyState__DetailsContainer"
    >
      <div
        class="Polaris-EmptyState__Details"
      >
        <div
          class="Polaris-TextContainer"
        >
          <p
            class="Polaris-DisplayText Polaris-DisplayText--sizeSmall"
          >
            No movies found
          </p>
          <div
            class="Polaris-EmptyState__Content"
          >
            <p>
              Search to get started!
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Polaris-EmptyState__ImageContainer"
    >
      <img
        alt=""
        class="Polaris-EmptyState__Image"
        role="presentation"
        src="https://www.iconarchive.com/download/i99782/designbolts/free-multimedia/Film.ico"
      />
    </div>
  </div>
</div>
`;
